### ISSUE TITLE
Complete Basis Set Migration

### ISSUE THREAD

**Issue by ndaelman-hu** [FAIRmat-NFDI/nomad-simulations#101]
Starting from `BasisSet(NumericalSettings)`, migrate over the basis set schema from `run` into `data`.

Semantically, the basis sets definitely belong under `NumericalSettings`. For clarity (towards the user), I'd prefer to keep this approach across the board. I have 2 questions before starting. Any input here is appreciated, but in particular from @JosePizarro3 :

1. Many basis sets refer to spatial components, e.g. atoms, atom groups, meshes. Is there a preferred way for connecting these? My initial suspicion would be _references_.
2. Basis sets are very modular. There is the collection as a whole, with different parts that each have their own parameters (split by space, energy, or Hamiltonian). The `run` schema had `BasisSetContainer` and `BasisSet` for each, respectively. I guess that the main function of `BasisSetContainer` is now taken care off by `ModelMethod.numerical_settings`.
    1. Firstly, is it fine to populate the repeating section `ModelMethod.numerical_settings` with all kinds of different basis set building blocks, or do we want more structure (e.g. a specific subsection like `BasisSetContainer`)?
    2. In case of the former option for 2.i, any suggestions on where to put the general basis set name (relevant for search) and `native_tier`?

For keeping track of the migration steps:
 - [x] base structure
 - [x] plane wave (excluding pseudopotentials)
 - [ ] atom-centered
 - [x] LAPW
 - [x] QuickStep

**Comment by JosePizarro3** [comment]
> 1. Many basis sets refer to spatial components, e.g. atoms, atom groups, meshes. Is there a preferred way for connecting these? My initial suspicion would be references.

Yeah, I would try out references, and see whether it makes sense.

> 2. Basis sets are very modular. There is the collection as a whole, with different parts that each have their own parameters (split by space, energy, or Hamiltonian). The run schema had BasisSetContainer and BasisSet for each, respectively. I guess that the main function of BasisSetContainer is now taken care off by ModelMethod.numerical_settings.
>     i. Firstly, is it fine to populate the repeating section ModelMethod.numerical_settings with all kinds of different basis set building blocks, or do we want more structure (e.g. a specific subsection like BasisSetContainer)?
>     ii. In case of the former option for 2.i, any suggestions on where to put the general basis set name (relevant for search) and native_tier?

I see, I think you might then end up using a placeholder like the one you have. I think you could still use the concept of `BasisSetContainer(NumericalSettings)` and add a repeated sub-section under it, something like:
```python
class BasisSetContainer(NumericalSettings):
    native_tier = Quantity()
    basis_set = SubSection(sub_section=BasisSet.m_def, repeats=True)
```

And here, `BasisSet` is an abstract class that plane waves, atom-centered, lapw, gaussians... all use to inherit. Thanks to this, you can add any of them as sub-sections of `BasisSetContainer`. And in each of the classes, you could specify basis-set-specific stuff (like references to AtomsStates or to Meshes). 

I think right now you had BasisSet as a class containing a lot of different sub-sections and quantities, all probably related with the different basis set types, right? I am not sure if I understand the fine details of things like `APW` and `APW+lo`, and whether the inheritance solution will help you.

**Comment by ndaelman-hu** [comment]
Closed by #102

**Comment by ndaelman-hu** [comment]
Atom-centered fucntions are a new feature to be added by C1 + C4
